### ISSUE TITLE
[RFC] Remove torch.compile dependency from Qwen3.8-Flash-Next NVIDIA path

### ISSUE THREAD

**Issue by gcanlin** [vllm-project/vllm#54688]
## Motivation

The NVIDIA implementation of Qwen3.8-Flash-Next (`Qwen4Exp` internally), introduced in #53896, still applies `support_torch_compile` to the main model and MTP classes. CUDA graph capture does not require this model-level AOT compilation, and the model-specific operations are increasingly implemented as explicit vLLM custom ops.

## Kernel list

`✅` means that the PR body contains measured performance results. Unit tests, accuracy evaluations, benchmark claims without data, and empty benchmark sections are not counted.

| Kernel | PR | Microbenchmark | E2E benchmark |
|---|---|:---:|:---:|
| QSA uniform decode/spec-decode indexer | #54513 | ✅ | ✅ |
| QSA varlen prefill indexer | #54513 | ✅ | ✅ |
| QSA block-index expansion | #54513 | ✅ | ✅ |
| PLE n-gram ID generation | #54517 | ✅ | ✅ |
| PLE varlen dilated convolution + SiLU/residual accumulation | #54517 | ✅ | ✅ |
| PLE convolution-state writeback | #54517 | ✅ | ✅ |
| PLE three-RMSNorm gate + convolution-input preparation | #54517 | ✅ | ✅ |
| MTP embedding residual add + first grouped Gemma RMSNorm | #54687 | ✅ | ✅ |
| PLE outer residual-add fusion | — | — | — |
| QSA sparse-attention + sigmoid output-gate epilogue | — | — | — |

### Eager / compiled baseline

```mermaid
flowchart TB
    E0["Token / multimodal embeddings<br/>[T, H]"] --> E1["Expand to four HC streams<br/>[T, 4H]"]

    subgraph EB["Backbone: 48 × Qwen4ExpDecoderLayer"]
        E1 --> E2["Decoder layer i<br/>i = 1...48"]
        E2 --> E3{"i = 2?"}
        E3 -- Yes --> E4["PLE sublayer<br/>torch n-gram ops → embedding lookup → KV projection<br/>three RMSNorms + gate → dense packing + F.conv1d + SiLU/state<br/>hidden_states + ple_output"]
        E3 -- No --> E5
        E4 --> E5["Attention HyperConnection<br/>grouped RMSNorm → down/injection GEMM → SiLU<br/>up GEMM → sigmoid gate reduction"]
        E5 --> E6{"Attention type"}
        E6 -- "36 linear-attention layers" --> E7["Gated DeltaNet<br/>linear-attention path"]
        E6 -- "Layers 4, 8, ..., 48" --> E8["QSA full attention<br/>QKV+gate projection → Q/K norm+RoPE<br/>shared indexer → sparse attention<br/>attn_output × sigmoid(output_gate) → O projection"]
        E7 --> E9
        E8 --> E9["MLP HyperConnection<br/>injection combine → grouped RMSNorm → down/injection GEMM<br/>SiLU → up GEMM → sigmoid gate reduction"]
        E9 --> E10["MLP / 512-expert MoE block"]
        E10 --> E11{"i < 48?"}
        E11 -- Yes --> E2
    end

    E11 -- No --> E12["Backbone final HyperConnection mixer<br/>consume final pending combine → grouped RMSNorm<br/>low-rank gate → collapse four streams"]
    E12 --> E13["Target sample hidden [T, H]<br/>→ LM head / target logits"]
    E12 --> E14["Backbone multi-stream hidden [T, 4H]"]
    E14 --> E15["MTP input preparation<br/>token embedding norm + projection<br/>hidden norm + projection<br/>embedding broadcast add → flatten → grouped RMSNorm"]
    E15 --> E16["MTP draft decoder layer × 1<br/>full-attention HyperConnection + QSA + MoE"]
    E16 --> E17["MTP final HyperConnection mixer<br/>collapse four streams"]
    E17 --> E18["Draft sample hidden [T, H]<br/>→ LM head / draft logits"]

    classDef eager fill:#ffe0b2,stroke:#e65100,color:#000
    class E0,E1,E2,E4,E5,E7,E8,E9,E10,E12,E13,E14,E15,E16,E17,E18 eager
```

### Ideal explicit / fused path

```mermaid
flowchart TB
    F0["Token / multimodal embeddings<br/>[T, H]"] --> F1["Four-stream HC state<br/>[T, 4H]"]

    subgraph FB["Backbone: 48 × Qwen4ExpDecoderLayer"]
        F1 --> F2["Decoder layer i<br/>i = 1...48"]
        F2 --> F3{"i = 2?"}
        F3 -- Yes --> F4["Fused PLE sublayer<br/>fused n-gram ID generation<br/>embedding lookup + KV projection<br/>fused three RMSNorms + gate + outer residual initialization<br/>varlen dilated depthwise conv + SiLU + state write + residual accumulation"]
        F3 -- No --> F5
        F4 --> F5["Fused Attention HyperConnection<br/>pending combine + grouped RMSNorm<br/>down/injection GEMM + SiLU epilogue<br/>up GEMM + sigmoid gate-reduction epilogue"]
        F5 --> F6{"Attention type"}
        F6 -- "36 linear-attention layers" --> F7["Gated DeltaNet<br/>explicit linear-attention kernels"]
        F6 -- "Layers 4, 8, ..., 48" --> F8["Fused QSA full attention<br/>packed QKV+gate projection<br/>fused Q/K norm + RoPE + cache write<br/>shape-specialized prefill/decode indexer<br/>sparse attention + sigmoid output-gate epilogue<br/>O projection"]
        F7 --> F9
        F8 --> F9["Fused MLP HyperConnection<br/>injection combine + grouped RMSNorm<br/>down/injection GEMM + SiLU epilogue<br/>up GEMM + sigmoid gate-reduction epilogue"]
        F9 --> F10["Fused MLP / MoE kernels"]
        F10 --> F11{"i < 48?"}
        F11 -- Yes --> F2
    end

    F11 -- No --> F12["Fused backbone final HC mixer<br/>pending combine + grouped RMSNorm<br/>low-rank projections + gate reduction<br/>produce sample and multi-stream hidden"]
    F12 --> F13["Target sample hidden [T, H]<br/>→ LM head / target logits"]
    F12 --> F14["Backbone multi-stream hidden [T, 4H]"]
    F14 --> F15["Fused MTP input<br/>embedding/hidden projections<br/>broadcast residual add + first grouped RMSNorm"]
    F15 --> F16["MTP draft decoder layer × 1<br/>reuse fused full-attention HC, QSA, and MoE paths"]
    F16 --> F17["Fused MTP final HC mixer<br/>produce next multi-stream state and sample hidden"]
    F17 --> F18["Draft sample hidden [T, H]<br/>→ LM head / draft logits"]

    classDef fused fill:#e8f5e9,stroke:#2e7d32,color:#000
    class F0,F1,F2,F4,F5,F7,F8,F9,F10,F12,F13,F14,F15,F16,F17,F18 fused
```

## CC List

cc @ZJY0516 @gau-nernst

**Comment by ZJY0516** [comment]
@gau-nernst will handle this, and I think we are close to it
